File: 3085_Minimum_Deletions_to_Make_String_K_Special.java
import java.util.Collections;
import java.util.HashMap;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

class Solution {
    public int minimumDeletions(String word, int k) {
        Map<Character, Integer> freqMap = new HashMap<>();
        for (char c : word.toCharArray()) {
            freqMap.put(c, freqMap.getOrDefault(c, 0) + 1);
        }

        List<Integer> uniqueFreqs = new ArrayList<>(freqMap.values());
        Collections.sort(uniqueFreqs);

        int minDeletions = Integer.MAX_VALUE;

        for (int lowFreqTarget : uniqueFreqs) {
            int currentDeletions = 0;
            for (int charFreq : freqMap.values()) {
                if (charFreq < lowFreqTarget) {
                    currentDeletions += charFreq;
                } else if (charFreq > lowFreqTarget + k) {
                    currentDeletions += charFreq - (lowFreqTarget + k);
                }
            }
            minDeletions = Math.min(minDeletions, currentDeletions);
        }
        
        return minDeletions;
    }
}

File: 3085_Minimum_Deletions_to_Make_String_K_Special.py
import collections

class Solution(object):
    def minimumDeletions(self, word, k):
        """
        :type word: str
        :type k: int
        :rtype: int
        """
        freq_map = collections.Counter(word)
        unique_freqs = sorted(list(freq_map.values()))
        min_deletions = float('inf')

        for low_freq_target in unique_freqs:
            current_deletions = 0
            for char_freq in freq_map.values():
                if char_freq < low_freq_target:
                    current_deletions += char_freq
                elif char_freq > low_freq_target + k:
                    current_deletions += char_freq - (low_freq_target + k)
            min_deletions = min(min_deletions, current_deletions)
            
        return min_deletions
